setup/generate_setup_file.py

```diff
@@ -0,0 +1,74 @@
+#! /usr/bin/env python3
+
+from typing import IO, List
+
+import os
+import sys
+import traceback
+
+
+class SetupGenerator:
+    def __init__(self, tue_env_dir=None):
+        if tue_env_dir is None:
+            tue_env_dir = os.environ["TUE_ENV_DIR"]
+        self._tue_env_dir = tue_env_dir
+        self._tue_dependencies_dir = os.path.join(self._tue_env_dir, ".env", "dependencies")
+        self._tue_env_targets_dir = os.environ["TUE_ENV_TARGETS_DIR"]
```

Comment:
raise exception if this is None

Reply:
Again, this will raise a `KeyError`, when not defined.